R/genetic-identification.R

```diff
@@ -240,49 +240,51 @@ ots_winter_spring_detection <- function(con, sample_id,
     dplyr::filter(sample_id == !!sample_id)
 
   assays_for_existing_for_sample <- assay_results |> dplyr::distinct(assay_id) |> dplyr::pull()
+  assays_3_for_sample <- assay_results |> dplyr::filter(assay_id == 3) |> dplyr::collect()
+  assays_4_for_sample <- assay_results |> dplyr::filter(assay_id == 4) |> dplyr::collect()
 
-  assay_needed_not_found <- !(3:4 %in% assays_for_existing_for_sample)
-
-  if (any(assay_needed_not_found)) {
+  if (nrow(assays_3_for_sample) == 0) {
+    cli::cli_warn(c(
+      "x" = "late and early assay needed to run early/late detection but assay = 3 was not found for sample {sample_id}"
```

Comment:
should this read `spring and winter assay needed to run spring/winter detection...` ? assay 3 is for spring/winter and not early/late